### ISSUE TITLE
Some problems about building the process for training

### ISSUE THREAD

**Issue by KennithLi** [fredfung007/snlt#13]
Thanks for your contribution. Could you provide the detailed environment config, data preparing, and training process? It is a little hard to follow the steps to train or test the codes. Some errors arise when I want to train the model:
`python run_experiment.py --num_machines=1 --config_file=configs.py `

```
Traceback (most recent call last):
  File "run_experiment.py", line 94, in <module>
    cfg.merge_from_file(FLAGS.config_file)
  File "/opt/conda/lib/python3.6/site-packages/yacs/config.py", line 212, in merge_from_file
    cfg = self.load_cfg(f)
  File "/opt/conda/lib/python3.6/site-packages/yacs/config.py", line 363, in load_cfg
    return cls._load_cfg_from_file(cfg_file_obj_or_str)
  File "/opt/conda/lib/python3.6/site-packages/yacs/config.py", line 374, in _load_cfg_from_file
    return cls._load_cfg_py_source(file_obj.name)
  File "/opt/conda/lib/python3.6/site-packages/yacs/config.py", line 393, in _load_cfg_py_source
    "Python module from file {} must have 'cfg' attr".format(filename),
  File "/opt/conda/lib/python3.6/site-packages/yacs/config.py", line 545, in _assert_with_logging
    assert cond, msg
AssertionError: Python module from file configs.py must have 'cfg' attr
````

**Comment by fredfung007** [comment]
The --config_file flag should be a relative path to one of the yaml files that are under the experiments/ folder.

**Comment by KennithLi** [comment]
Thank you for your reply. I am still confused by the environment to run the codes. Which version of the torch is used in the project? I want to run it with torch==0.4.0 or 0.4.1, and they both fail to run. 

torch==0.4.0, errors:
```
Traceback (most recent call last):
  File "run_experiment.py", line 18, in <module>
    from tracker.siamrpnpp_multinlrpn import SiamRPNPPMultiNLRPN
  File "snlt/tracker/siamrpnpp_multinlrpn.py", line 6, in <module>
    from region_proposal_network.multi_nl_rpn import MultiNLRPN
  File "snlt/region_proposal_network/multi_nl_rpn.py", line 3, in <module>
    from transformers import BertTokenizer, BertModel
  File "<frozen importlib._bootstrap>", line 1020, in _handle_fromlist
  File "/opt/conda/envs/pytorch-py3.6/lib/python3.6/site-packages/transformers/file_utils.py", line 2026, in __getattr__
    module = self._get_module(self._class_to_module[name])
  File "/opt/conda/envs/pytorch-py3.6/lib/python3.6/site-packages/transformers/file_utils.py", line 2035, in _get_module
    return importlib.import_module("." + module_name, self.__name__)
  File "/opt/conda/envs/pytorch-py3.6/lib/python3.6/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "/opt/conda/envs/pytorch-py3.6/lib/python3.6/site-packages/transformers/models/__init__.py", line 19, in <module>
    from . import (
  File "/opt/conda/envs/pytorch-py3.6/lib/python3.6/site-packages/transformers/models/layoutlm/__init__.py", line 22, in <module>
    from .configuration_layoutlm import LAYOUTLM_PRETRAINED_CONFIG_ARCHIVE_MAP, LayoutLMConfig
  File "/opt/conda/envs/pytorch-py3.6/lib/python3.6/site-packages/transformers/models/layoutlm/configuration_layoutlm.py", line 19, in <module>
    from ..bert.configuration_bert import BertConfig
  File "/opt/conda/envs/pytorch-py3.6/lib/python3.6/site-packages/transformers/models/bert/configuration_bert.py", line 21, in <module>
    from ...onnx import OnnxConfig
  File "/opt/conda/envs/pytorch-py3.6/lib/python3.6/site-packages/transformers/onnx/__init__.py", line 17, in <module>
    from .convert import export, validate_model_outputs
  File "/opt/conda/envs/pytorch-py3.6/lib/python3.6/site-packages/transformers/onnx/convert.py", line 23, in <module>
    from .. import PreTrainedModel, PreTrainedTokenizer, TensorType, TFPreTrainedModel, is_torch_available
  File "<frozen importlib._bootstrap>", line 1020, in _handle_fromlist
  File "/opt/conda/envs/pytorch-py3.6/lib/python3.6/site-packages/transformers/file_utils.py", line 2026, in __getattr__
    module = self._get_module(self._class_to_module[name])
  File "/opt/conda/envs/pytorch-py3.6/lib/python3.6/site-packages/transformers/file_utils.py", line 2035, in _get_module
    return importlib.import_module("." + module_name, self.__name__)
  File "/opt/conda/envs/pytorch-py3.6/lib/python3.6/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "/opt/conda/envs/pytorch-py3.6/lib/python3.6/site-packages/transformers/modeling_utils.py", line 47, in <module>
    from .generation_utils import GenerationMixin
  File "/opt/conda/envs/pytorch-py3.6/lib/python3.6/site-packages/transformers/generation_utils.py", line 363, in <module>
    class GenerationMixin:
  File "/opt/conda/envs/pytorch-py3.6/lib/python3.6/site-packages/transformers/generation_utils.py", line 680, in GenerationMixin
    ) -> Union[GreedySearchOutput, SampleOutput, BeamSearchOutput, BeamSampleOutput, torch.LongTensor]:
TypeError: 'no_grad' object is not callable
```

torch==0.4.1, errors:
```
  File "run_experiment.py", line 19, in <module>
    from tracker.trainer import TrackerTrainer
  File "snlt/tracker/trainer.py", line 8, in <module>
    from torch.utils.data import DistributedSampler, DataLoader
ImportError: cannot import name 'DistributedSampler'
```

**Comment by yepingapril** [comment]
> Thanks for your contribution. Could you provide the detailed environment config, data preparing, and training process? It is a little hard to follow the steps to train or test the codes. Some errors arise when I want to train the model: `python run_experiment.py --num_machines=1 --config_file=configs.py `
> 
> ```
> Traceback (most recent call last):
>   File "run_experiment.py", line 94, in <module>
>     cfg.merge_from_file(FLAGS.config_file)
>   File "/opt/conda/lib/python3.6/site-packages/yacs/config.py", line 212, in merge_from_file
>     cfg = self.load_cfg(f)
>   File "/opt/conda/lib/python3.6/site-packages/yacs/config.py", line 363, in load_cfg
>     return cls._load_cfg_from_file(cfg_file_obj_or_str)
>   File "/opt/conda/lib/python3.6/site-packages/yacs/config.py", line 374, in _load_cfg_from_file
>     return cls._load_cfg_py_source(file_obj.name)
>   File "/opt/conda/lib/python3.6/site-packages/yacs/config.py", line 393, in _load_cfg_py_source
>     "Python module from file {} must have 'cfg' attr".format(filename),
>   File "/opt/conda/lib/python3.6/site-packages/yacs/config.py", line 545, in _assert_with_logging
>     assert cond, msg
> AssertionError: Python module from file configs.py must have 'cfg' attr
> ```

I have the same error as you when train the codes. There is no experiments/test.yaml in the folder. How to solve the problem in the end?
